### DISCUSSION TITLE
Blazor wasm problem on .NET 5+: PlatformNotSupportedException

### DISCUSSION THREAD

**Discussion by horacioj** [pnp/pnpcore#705] [Q&A] [answered]
Hello,

The [Blazor WebAssembly sample](https://github.com/pnp/pnpcore/tree/dev/samples/Demo.Blazor) works fine as it is.
However, it crashes if for example I simply update SiteInfo.razor to get a list of pages: `var pages = await ctx.Site.RootWeb.GetPagesAsync();`

```
Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: The type initializer for 'PnP.Core.Model.SharePoint.Page' threw an exception.
System.TypeInitializationException: The type initializer for 'PnP.Core.Model.SharePoint.Page' threw an exception.
 ---> System.PlatformNotSupportedException: System.Security.Cryptography.Algorithms is not supported on this platform.
   at System.Security.Cryptography.MD5.Create()
   at PnP.Core.Model.SharePoint.Page..cctor()
   --- End of inner exception stack trace ---
   at PnP.Core.Model.SharePoint.Web.GetPagesAsync(String pageName)
   at Demo.Blazor.Pages.SiteInfo.OnInitializedAsync() in F:\trabajo\source\pnpcore-blazor-webassembly\Pages\SiteInfo.razor:line 32
   at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)
```

I'm targeting .NET 5, and apparently this exception cannot be solved: https://docs.microsoft.com/en-us/dotnet/core/compatibility/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly

> In previous .NET versions, most of the System.Security.Cryptography APIs aren't available to Blazor WebAssembly apps. Starting in .NET 5, Blazor WebAssembly apps target the full .NET 5 API surface area, however, not all .NET 5 APIs are supported due to browser sandbox constraints. In .NET 5 and later versions, the unsupported System.Security.Cryptography APIs throw a PlatformNotSupportedException when running on WebAssembly

Do you know know can I make this work on .NET 5+? 

Alternatively (or perhaps, even better...) do you have an example of the same thing (Blazor sample) but as a Blazor Server project? 

I tried to create a basic Blazor Server example, but I couldn't make the authentication work. 
Would it be possible to get an example of how the authentication should be implemented when using Blazor Server? 

Many thanks!

**Comment by jansenbe** [comment, marked as answer]
Thanks for bringing this up @horacioj , the MD5 usage is coming from an experimental feature (creating SharePoint Topic pages) which is an  edge case that will not be used by many. For now I'm rewriting the code so that things can work in Blazor web assembly without needing MD5.

**Reply by jansenbe** [reply]
@horacioj : problem is fixed in the current nightly release. Also updated the Blazor sample (https://github.com/pnp/pnpcore/tree/dev/samples/Demo.Blazor) and added a razor page that loads the site pages. Let me know if you encounter other Blazor issues.

**Reply by horacioj** [reply]
Thanks a lot, @jansenbe!
